src/GraphQL/Internal/AST.hs

```diff
@@ -1,9 +1,11 @@
 module GraphQL.Internal.AST
   ( Name
+  , pName
```

Comment:
would you mind renaming to `parseName` or similar? Took me a while to figure out what you mean here.